packages/suite/src/actions/wallet/stakeActions.ts

```diff
@@ -0,0 +1,213 @@
+import BigNumber from 'bignumber.js';
```

Comment:
This file looks really similar to send form actions. 

However, let's leave it like that to make it easier for now and decide later based on how it will be different from send form.

I am really afraid that leaving it almost similar could cause us really big troubles in the future as we plan to create "standalone app" for send form logic. 

You are also extending the sendForm types and using it, so as I say, lets leave it like that and it should not be that hard to change it later. I hope

Reply:
It 100% looks similar. It was my intention in order to keep the same approach and not reinvent the wheel. + it helps when using it in the `TransactionReviewModal`. This way, we don’t have to add any additional type guards, checks etc. there.

Couldn’t know about you future plans about extraction the send logic. If you have a concrete vision of how it should be designed, please, share it with me and I’ll try of rewrite it like you want. imho, as I mentioned in the PR description, this logic should be at least in the thunks file in `wallet-core`

I do extend `sendForm` types. I’ve seen you do it in your PR, that’s why I thought it would be OK. + this `FormState` type is used across all the forms in the app, which also made me think it would be OK. Would you like a completely separate form type for staking?

Reply:
Yes please, create a separate FormState just for staking

Reply:
Created a completely separate `StakeFormState`. It has majority of values from send `FormState` excluding bitcoin-oriented ones.

This change would require mixing different form states in `useCompose` and `sendFormEthereumActions`. Since, you’re planning to extract everything concerning send logic in the future I’ve decided to create separate versions of those logic pieces: `useStakeCompose` and `stakeFormEthereumActions` respectively. Those files deal with `StakeFormState` and stake-related types exclusively and I removed some logic, which seemed not needed there, e.g. tokens logic.

Also, since there’s only one review modal for all operations in the app, I had to modify `TransactionReviewModal` a bit to make it work with different form states and satisfy TS compiler.

https://github.com/everstake/trezor-suite/pull/4/commits/be670f99338d06d65807c8520b20b9c07fe3a049